src/Types/Saber/Saber.cpp

```diff
@@ -49,19 +50,35 @@ namespace Qosmetics
             saberType = gameSaber->get_saberType();
 
         modelController = GetComponentInChildren<GlobalNamespace::SaberModelController*>(true);
+
+        // Pass in through constructor to avoid redoing?
+        GlobalNamespace::SaberManager *saberManager = FindObjectOfType<GlobalNamespace::SaberManager *>();
+
+        if (saberManager) {
+            auto otherSaber = saberType.value == 0 ? saberManager->leftSaber : saberManager->rightSaber;
+            if (otherSaber) {
+                otherModelController = otherSaber->GetComponentInChildren<GlobalNamespace::SaberModelController *>(true);
+            }
+        }
+        INFO("Going to register Chroma! %i", saberType.value);
         if (modelManager->get_item().get_type() != ItemType::invalid)
         {
+            INFO("Going to register Chroma! v2 %i", saberType.value);
             ChromaUtils::setSaberColoredByChroma(modelController, false);
             ChromaUtils::registerSaberCallback({&Saber::UpdateChromaColors, this});
         }
-        
+
+        Il2CppString *saberName = (saberType.value == 1) ? modelManager->get_leftSaberName() : modelManager->get_rightSaberName();
+        modelObject = get_transform()->Find(saberName);
```

Comment:
This will never find the model as this is always before the model actually gets replaced